src/services/item/plugins/published/repositories/itemPublished.ts

```diff
@@ -38,64 +55,91 @@ export const ItemPublishedRepository = AppDataSource.getRepository(ItemPublished
         entries.find((e) => items.find((i) => i.id === id)?.path.startsWith(e.item.path)),
       buildError: (id) => new ItemPublishedNotFound(id),
     });
-  },
+  }
+
+  async getForMember(memberId: Member['id']): Promise<Item[]> {
+    const itemPublished = await this.repository
+      .createQueryBuilder('pi')
+      .innerJoin('item_membership', 'im', 'im.item_path @> pi.item_path')
+      .innerJoin('member', 'm', 'im.member_id = m.id and m.id = :memberId', {
+        memberId,
+      })
```

Comment:
If it's what we discussed, you don't use a `where` clause because it is only computed at the end of the query, right? Maybe add a comment.

Reply:
Yes exactly, I add a condition in the join to minimize the number of rows that will be joined. I added the same for the memberships.